### PR TITLE
Don't override student causal mask

### PR DESCRIPTION
**Pull request by gsganden**
`tests/test_attention_block.py::test_eager_causal_attention_block` is currently overriding the student model's causal mask, which can give false negatives when the student's causal mask is wrong and false positives when the student's causal masking is implemented differently (e.g. using `False` to indicate masked positions, appropriately coordinated with `forward`).

### CODE REVIEW

**Comment by warner-benjamin** [comment]
Good catch. I should have added `persistent=False` to `register_buffer` which prevents it from being in the `state_dict` (done in  ab35969) and then I improved loading and a custom error message in d661e58.

```python
self.register_buffer(
    "causal_mask",
    torch.triu(torch.ones(max_seq_len, max_seq_len, dtype=torch.bool), diagonal=1),
    persistent=False,
)
```
